src/objects.c

```diff
@@ -963,17 +963,16 @@ void disconnect_server(PgSocket *server, bool send_term, const char *reason, ...
 }
 
 /*
- * close client connection
+ * A wrapper around disconnect_client_sqlstate()
  *
- * notify=true means to send the reason message as an error to the
- * client, notify=false means no message is sent.  The latter is for
- * protocol and communication errors where sending a regular error
- * message is not possible.
+ * The function disconnect_client_sqlstate() inherits the disconnect_client()
+ * content and add a new option that provides a specific SQLSTATE that is
+ * forwarded to client.  PgBouncer used to report SQLSTATE 08P01
+ * (protocol_violation) for all cases but it diverges from what Postgres
+ * reports in some cases.
  */
 void disconnect_client(PgSocket *client, bool notify, const char *reason, ...)
 {
-	usec_t now = get_cached_time();
-
 	if (reason) {
```

Comment:
I think this should be moved to `disconnect_client_sqlstate`.

Reply:
Are you referring to `now` variable? It was moved to `disconnect_client_sqlstate`. I checked a few times after reading the diff. diff seems confusing. `now` is in line 999.

Reply:
no I meant the `if (reason) {` block